app/controllers/users_controller.rb

```diff
@@ -16,7 +16,22 @@ def create
     end
   end
 
-  def show
+  def edit
+    if User.find(params[:id]).admin?
+      @user = current_user
+    else
+      render file: 'app/views/errors/not_found.html.erb'
+    end
+  end
+
+  def update
+    @user = current_user
+    if @user.update_attributes(user_params)
```

Comment:
I think we could this in one less line if we used `if current_user.update_attributes()`. Just a thought in the way of a skinny controller